ci/tiledb-java-linux_osx-release.yml

```diff
@@ -0,0 +1,68 @@
+steps:
+  - bash: |
+      set -e pipefail
+      open /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg
+      sudo installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -allowUntrusted -target /
+    condition: eq(variables['Agent.OS'], 'Darwin')
+    displayName: 'Install system headers (OSX only)'
+
+  - bash: |
+      unset SYSTEM
+      ls
+
+      BUILD_BINARIESDIRECTORY=${BUILD_BINARIESDIRECTORY:-$BUILD_REPOSITORY_LOCALPATH/}
+
+      ./gradlew assemble
+
+      if [[ ( "$AGENT_OS" == "Linux" ) ]]; then
+            cp ./build/tiledb_jni/*.so ./build/install/lib/*.so $BUILD_BINARIESDIRECTORY
+      fi
+
+      if [[ ( "$AGENT_OS" == "Darwin" ) ]]; then
+              cp ./build/tiledb_jni/*.dylib ./build/install/lib/*.dylib $BUILD_BINARIESDIRECTORY
+      fi
+
+      echo "Native Libs"
+      ls $BUILD_BINARIESDIRECTORY
+
+    displayName: 'Build and Upload (Ubuntu and OSX)'
+
+  - bash: |
+      set -e pipefail
+      # Display log files if the build failed
+      echo "Dumping log files for failed build"
+      echo "----------------------------------"
+      for f in $(find $BUILD_REPOSITORY_LOCALPATH -name *.log);
+        do echo "------"
+           echo $f
+           echo "======"
+           cat $f
+        done;
+    condition: failed() # only run this job if the build step failed
+    displayName: "Print log files (failed build only)"
+
+
+  - script: |
+      echo $sourceVersion
+      commitHash=${sourceVersion:0:7}
+      echo $commitHash
+      echo "##vso[task.setvariable variable=commitHash]$commitHash" ## Set variable for using in other tasks.
+    env: { sourceVersion: $(Build.SourceVersion) }
+    displayName: Git Hash 7-digit
+
+  - task: ArchiveFiles@2
+    inputs:
+      rootFolderOrFile: '$(Build.BinariesDirectory)'
+      includeRootFolder: false
+      archiveType: 'tar' # Options: zip, 7z, tar, wim
+      tarCompression: 'gz' # Optional. Options: gz, bz2, xz, none
+      archiveFile: $(Build.ArtifactStagingDirectory)/tiledb-$(Agent.OS)-$(Build.SourceBranchName)-$(commitHash).tar.gz
+      replaceExistingArchive: true
+      verbose: true # Optional
+    condition: succeeded()
+
+  - task: PublishBuildArtifacts@1
+    inputs:
+      pathtoPublish: $(Build.ArtifactStagingDirectory)/tiledb-$(Agent.OS)-$(Build.SourceBranchName)-$(commitHash).tar.gz
+      artifactName: tiledb-$(Agent.OS)-$(Build.SourceBranchName)-$(commitHash).tar.gz
```

Comment:
you can use one common artifact name for all libraries. Something like `libraries`.